plugins/out_stackdriver/stackdriver.c

```diff
@@ -33,6 +34,65 @@
 #include <mbedtls/base64.h>
 #include <mbedtls/sha256.h>
 
+/* Validate resource labels against required labels for a monitored resource */
+static int validate_resource_labels(struct flb_stackdriver *ctx)
```

Comment:
It's a bit weird to see the list of `valid_resources` in `stackdriver_conf.c`, but the `vaildate_resource_labels` function in `stackdriver.c`. If would be nice to consolidate all resource-related functions in one file. If it can't be either `stackdriver.c` or `stackdriver_conf.c`, maybe `resource.c`?

Reply:
Yeah, I am breaking the resources out into resource.h/c with associated methods.